### ISSUE TITLE
It's not working with real-time high dimensitional dataset

### ISSUE THREAD

**Issue by shwetashrm-1994** [omadson/fuzzy-c-means#31]
I am trying to apply fuzzy c-means algorithm to cluster out my high-dimensional dataset but its not working please suggest some solution as I need to cluster data to solve class-imbalance
@omadson

**Comment by afparsons** [comment]
Please follow general GitHub etiquitte.

- write in complete sentences and not avoid writing run-on sentences. Grammatical mistakes for non-native speakers of English  are of course understandable, so long as an honest effort made (we are all lifelong learners in one capacity or another :slightly_smiling_face:).
- more importantly, when requesting help, it is expected that users provide example code, example outputs, and a description of expected output. When relevant, it can be helpful to specify system information and pertinent package versions.

**Comment by VaishU2235** [comment]
I faced a similar problem today while trying to run fuzzy c means clustering on higher-dimensional space (dimension > 2000) with more than 50 observations. I tried to find 3-15 clusters and every time fuzzy c means gave me 2 clusters in the result. It is an inherent issue of Fuzzy C Means as described in this [paper](https://www.researchgate.net/publication/225005392_Fuzzy_C-Means_in_High_Dimensional_Spaces). Is there a solution for this kind of problem?
